### PR TITLE
Add custom icon support to sidebar

### PR DESCRIPTION
**Pull request by dnewh**
This adds support for custom icons for a sapling in the sidebar. These
custom icons are specified by setting the `logo` field in a sapling's
entry in the sapling manifest with a URL to the image. The custom image
takes precedence over an icon set with the `icon` field but at least one
of these fields is required.

Signed-off-by: Davey Newhall <newhall@bitwise.io>

### CODE REVIEW

**Review by isabeltomb — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "@babel/cli": "^7.8.4",
     "@babel/core": "^7.8.4",
     "@babel/plugin-transform-runtime": "^7.8.3",
-    "@babel/preset-env": "^7.8.4",
+    "@babel/preset-env": "^7.8.7",
```
Comment (peterschwarz) [inline package.json:49]: Could you split this change into a separate commit and explain the change?

File: src/components/navigation/NavItem.js [hunk 2]
```diff
@@ -30,16 +30,36 @@ export const NavItem = props => {
...
+          )}
         </div>
       </div>
       <div className="label">{label}</div>
     </a>
   );
 };
 
+const iconImagePropsCheck = (props, propName, componentName) => {
+  if (!props.logo && !props.brandLogo) {
```
Comment (peterschwarz) [inline src/components/navigation/NavItem.js]: This removes the type check that requires them to be strings

**Review by peterschwarz — CHANGES_REQUESTED** [review]
I'm not sure why this `brandLogo` isn't just passed as `logo`, especially since they are mutually exclusive. Perhaps this change the logo type to be an element, instead of a field.

File: package.json (hunk 1 above)
Comment (dnewh) [reply package.json:49]: done

File: src/components/navigation/NavItem.js (hunk 2 above)
Comment (dnewh) [reply src/components/navigation/NavItem.js]: fixed

**Review by peterschwarz — APPROVED** [review]

**Review by isabeltomb — APPROVED** [review]
